dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -109,24 +114,52 @@ private PredicateBuilder initiatorPredicateBuilder(long initiatorUserId) {
 
     private PredicateBuilder accessPredicateBuilder(User loggedInUser) {
         return (cb, event) -> {
-            // Calculate a predicate representing public access.
-            // Currently, this is all events that don't refer to a "categorizer" organization.
-            Join<Event, Organization> organization = event.join("organization", JoinType.LEFT);
-            Predicate publicPredicate = cb.or(cb.isFalse(organization.get("categorizer")), cb.isNull(organization));
-            // Calculate a predicate representing extra access due to the user's identity.
-            Predicate extraPredicate;
             if (loggedInUser != null && (loggedInUser.getIsAdmin() || loggedInUser.isCurator())) {
-                // Admin or curator, able to access everything.
-                extraPredicate = cb.conjunction();
-            } else if (loggedInUser != null) {
-                // Logged in, able to access events that the user initiated.
-                extraPredicate = event.get("initiatorUser").in(loggedInUser.getId());
+                // An admin or curator is able to access everything.
+                return cb.conjunction();
+            }
+            // Calculate a predicate representing public access, which, currently, is all events that:
+            // 1) Either:
+            // a) Don't refer to an entry
+            // b) Refer to a published entry, or
```

Comment:
This is a little hard to wrap my head around; does this mean you'll be able to see events all published entries, are just the ones you've starred?

Reply:
This function builds a database predicate that defines everything the logged-in user is allowed to see in the events table, given no other restrictions.  The  `findEvents` function combines it with another predicate `specifiedPredicate` that's computed from the specified predicate builder, which defines the subset of events that the caller wants to see, for example, events that were initiated by a certain user.  And, all of other methods that return a list of events use `findEvents` internally.